lib/utils/httpRequestor.js

```diff
@@ -99,59 +95,71 @@ exports.create = function(requestorConfig) {
-  };
-
-  var methodRequest = (options, method, methodName, callback, requestExtension) => {
+  var methodRequest = (options, method, methodName, callback, body) => {
     var baseRequestOptions = {
       url: buildUrl(options),
       headers: buildHeaders(options),
-      qs: options.queryParameters,
-      encoding: options.encoding,
-      gzip: true
+      params: options.queryParameters,
+      responseEncoding: options.encoding
     };
-    var requestOptions = _.extend(baseRequestOptions, requestExtension);
+    var url = buildUrl(options);
+    var requestOptions = baseRequestOptions;
 
     var retryOptions = _.pick(options, 'maxRetryDurationMillis', 'calcRetryBackoff');
 
     logger.logRequest(methodName, requestOptions);
 
-    return makeRequestWithRetries(method, methodName, requestOptions, retryOptions, callback);
+    return makeRequestWithRetries(url, method, methodName, requestOptions, body, retryOptions, callback);
   };
 
-  var makeRequestWithRetries = (method, methodName, requestOptions, retryOptions, callback) => {
+  var makeRequestWithRetries = (url, method, methodName, requestOptions, body, retryOptions, callback) => {
     var effectiveRetryOptions = _.defaults(retryOptions, defaultRetryOptions);
 
     effectiveRetryOptions.endRetryTime = Date.now() + effectiveRetryOptions.maxRetryDurationMillis;
 
-    return retryHelper(method, methodName, requestOptions, effectiveRetryOptions, 0)
-      .tap(logger.logSuccessfulResponse)
-      .tapCatch(error => logger.logErrorResponse(methodName, requestOptions, error))
-      .get('content')
-      .catch(error => Promise.reject(_.omit(error, 'headers', 'body')))
-      .nodeify(callback);
+    return retryHelper(url, method, methodName, requestOptions, body, effectiveRetryOptions, 0)
+      .then((response) => {
+        logger.logSuccessfulResponse(response);
+
+        if (callback) {
+          callback(undefined, response.content);
+        }
+        
+        return response.content;
+      })
+      .catch((error) => {
+        logger.logErrorResponse(methodName, requestOptions, error);
+
+        if (callback) {
+          callback(error, undefined);
+        }
+        
+        return new Promise.reject(_.omit(error, 'headers', 'body'));
+      });
   };
 
-  var retryHelper = (method, methodName, requestOptions, retryOptions, numRetries) =>
-    makeRequest(method, methodName, requestOptions)
-      .catch(retryWithBackoffHelper(method, methodName, requestOptions, retryOptions, numRetries));
+  var retryHelper = (url, method, methodName, requestOptions, body, retryOptions, numRetries) => {
+    return methodHandler(url, method, methodName, requestOptions, body)
+      .then(handleResponse)
+      .catch(retryWithBackoffHelper(url, method, methodName, requestOptions, retryOptions, numRetries));
+  };
+  
+  var methodHandler = (url, method, methodName, requestOptions, body) => {
+    if (methodName === "POST" || methodName === "PUT") {
```

Comment:
Has there been any test(s) added to cover the regression scenario we hit with the last PR?

Reply:
Yes there is test coverage to ensure this logic is functional. Originally there was coverage to catch the contract with the http request library but the issue surfaced from incorrectly updating those tests 😅. When I added the fix for this issue there were 3 tests failing for `POST` and 3 tests failing for `PUT` that I updated to reflect the correct `axios` contract so we should be covered again moving forward 😄